portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -22,44 +33,60 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some placeholder comment saying 'Hello, Omar!'. TODO(oumontiel): modify this file to handle comments data */
+/** Servlet that posts and retrieves comments from Datastore. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
   /**
-   * List of messages to be fetched from this Data Servlet
-   * @private {ArrayList<String>}
+   * Gets the comments, in JSON format, taken from user input in the form from the HTML,
+   * which are stored in the messages variable.
    */
-  private ArrayList<String> messages;
-
-  @Override
-  public void init() {
-    messages = new ArrayList<>();
-    messages.add("Once upon a time...");
-    messages.add("Twice upon two times...");
-    messages.add("Thrice upon three times...");
-  }
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    /** Create a query and prepare it with the data stored in Datastore. */
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    /**
+     * Add all queried comments from Datastore
+     * to an List of type Comment.
+     */
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      String content = (String) entity.getProperty("content");
+      long timestamp = (long) entity.getProperty("timestamp");
```

Comment:
Is there ever a chance that any of these values could be null? If they were null, what kind of behavior would exist? It may be more robust to do some sort of error checking here!

Reply:
They can't be null. If anything, content could be an empty string, but that wouldn't generate any type of error, it would just display an empty string on the webpage. Is this ok?

Reply:
Gotcha, yeah that's fine - just wanted to make sure you had considered various fail states!